prql-compiler/tests/integration/main.rs

```diff
@@ -1,20 +1,43 @@
 // Re-enable on windows when duckdb supports it
 // https://github.com/wangfenjin/duckdb-rs/issues/62
 #![cfg(not(any(target_family = "windows", target_family = "wasm")))]
+// TODO enable it for all OS
+#![cfg(target_os = "linux")]
+#![cfg(feature = "test-external-dbs")]
+
+mod connection;
 
 #[cfg(test)]
 mod tests {
-    use std::{fs, path::Path};
+    use std::collections::BTreeMap;
+    use std::fmt::Write;
+    use std::{env, fs};
 
     use insta::{assert_snapshot, glob};
-    use prql_compiler::{sql::Dialect, Options, Target};
+    use postgres::NoTls;
+    use tiberius::{AuthMethod, Client, Config};
+    use tokio::net::TcpStream;
+    use tokio::runtime::Runtime;
+    use tokio_util::compat::{Compat, TokioAsyncWriteCompatExt};
+
+    use prql_compiler::sql::Dialect;
+    use prql_compiler::Options;
+    use prql_compiler::Target::Sql;
+
+    use crate::connection::*;
 
     #[test]
-    fn test() {
-        // TODO: we could have a trait rather than mods for each of these?
-        let mut pg_client = postgres::connect();
-        let duckdb_conn = duckdb::connect();
-        let sqlite_conn = sqlite::connect();
+    fn test_rdbms() {
+        if env::var("SKIP_INTEGRATION").is_ok() {
```

Comment:
(Fine to leave for a different PR)

I was thinking we would remove this, and only rely on whether the feature was enabled, so there's no external context leaking in here. I'll add a comment on how I was thinking about doing that in the CI file in case that's helpful.